test/helpers/kubectl.go

```diff
@@ -344,47 +344,42 @@ func (kub *Kubectl) Logs(namespace string, pod string) *CmdRes {
 // a timeout. Also it returns a callback function to stop the monitor and save
 // the output to `helpers.monitorLogFileName` file.
 func (kub *Kubectl) MicroscopeStart() (error, func() error) {
-	/*
-		TODO: re-enable microscope in CI. See GH-4011.
-		microscope := "microscope"
-		cmd := fmt.Sprintf("%[1]s -n %[2]s exec %[3]s -- %[3]s",
-			KubectlCmd, KubeSystemNamespace, microscope)
-		_ = kub.Apply(microscopeManifest)
-
-		_, err := kub.WaitforPods(
-			KubeSystemNamespace,
-			fmt.Sprintf("-l k8s-app=%s", microscope),
-			300)
-		if err != nil {
-			return err, nil
-		}
+	microscope := "microscope"
+	cmd := fmt.Sprintf("%[1]s -n %[2]s exec %[3]s -- %[3]s",
+		KubectlCmd, KubeSystemNamespace, microscope)
+	_ = kub.Apply(microscopeManifest)
+
+	err := kub.WaitforPods(
+		KubeSystemNamespace,
+		fmt.Sprintf("-l k8s-app=%s", microscope),
+		300)
+	if err != nil {
+		return err, nil
+	}
 
-		ctx, cancel := context.WithCancel(context.Background())
-		res := kub.ExecContext(ctx, cmd, ExecOptions{SkipLog: true})
+	ctx, cancel := context.WithCancel(context.Background())
+	res := kub.ExecContext(ctx, cmd, ExecOptions{SkipLog: true})
 
-		cb := func() error {
-			cancel()
-			testPath, err := CreateReportDirectory()
-			if err != nil {
-				kub.logger.WithError(err).Errorf(
-					"cannot create test results path '%s'", testPath)
-				return err
-			}
+	cb := func() error {
+		cancel()
+		testPath, err := CreateReportDirectory()
+		if err != nil {
+			kub.logger.WithError(err).Errorf(
+				"cannot create test results path '%s'", testPath)
+			return err
+		}
 
-			err = ioutil.WriteFile(
-				filepath.Join(testPath, monitorLogFileName),
-				res.CombineOutput().Bytes(),
-				LogPerm)
-			if err != nil {
-				log.WithError(err).Errorf("cannot create monitor log file")
-			}
-			return nil
+		err = ioutil.WriteFile(
+			filepath.Join(testPath, monitorLogFileName),
+			res.CombineOutput().Bytes(),
+			LogPerm)
+		if err != nil {
+			log.WithError(err).Errorf("cannot create monitor log file")
```

Comment:
Ouch, could you add a return err here please?

Reply:
done

Reply:
Thanks!